xarray/core/indexing.py

```diff
@@ -1417,41 +1430,54 @@ def __init__(self, array):
         """
         self.array = array
 
+    def _rewrite_indexer(self, key):
```

Comment:
Let's delete this and close https://github.com/pydata/xarray/issues/8377. There will be a failing test around `.sort` that can also just be deleted.

Reply:
Done: 

- https://github.com/pydata/xarray/pull/8758